dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/BratWriter.java

```diff
@@ -604,27 +652,57 @@ private void writeTextAnnotation(BratAnnotationDocument aDoc, AnnotationFS aFS)
-        nextTextAnnotationId++;
-
-        conf.addEntityDecl(superType, type);
-        
-        conf.addLabelDecl(anno.getType(), aFS.getType().getShortName(), aFS.getType()
-                .getShortName().substring(0, 1));
-
-        if (!conf.hasDrawingDecl(anno.getType())) {
-            conf.addDrawingDecl(new BratTextAnnotationDrawingDecl(anno.getType(), "black",
-                    palette[nextPaletteIndex % palette.length]));
-            nextPaletteIndex++;
+        // check if the type has  a value to display, replace the the type by the value
+        // do it similar as with declarations that looks for the value
+        LinkedList<String> spanDatas=null;
+        String value=type;
+        if (spanTypes.containsKey(aFS.getType().getName())) {
+        	spanDatas=spanTypes.get(aFS.getType().getName());
+        } else  if (spanTypes.containsKey(aFS.getCAS().getTypeSystem().getParent(aFS.getType()).getName() ))  {
+        	spanDatas=spanTypes.get(aFS.getCAS().getTypeSystem().getParent(aFS.getType()).getName() );
+        }
+        for (String spanData : spanDatas){	
+	       try {
+	        	if (!spanData.equalsIgnoreCase("")){
+	        		String [] splits=spanData.split("\\|");
+	        		if (splits.length>1){
+	        			FeatureStructure currentAnnot = aFS.getFeatureValue(aFS.getType().getFeatureByBaseName(splits[0]));
+	        			for (int f=1;f<splits.length-1;f++){     		
+		        		currentAnnot = currentAnnot.getFeatureValue(currentAnnot.getType().getFeatureByBaseName(splits[f]));
+		        	}
+		        	value= currentAnnot.getFeatureValueAsString(currentAnnot.getType().getFeatureByBaseName(splits[splits.length-1]));
+		        }  	else {
+		        	value=aFS.getFeatureValueAsString(aFS.getType().getFeatureByBaseName(splits[0]));
+		        	}
+		        }
+	        } catch (NullPointerException E){ // if the feature is not there, then a null pointer exception raises. 
+	        	getLogger().error("Type "+ type + "does not have the fature: "+ spanData + "as defined in PARAM_TEXT_ANNOTATION_TYPES"); 
+	        }  
+ 
+		    BratTextAnnotation anno = new BratTextAnnotation(nextTextAnnotationId, value,
+		            aFS.getBegin(), aFS.getEnd(), aFS.getCoveredText());
+		    nextTextAnnotationId++;
+		
+		    conf.addEntityDecl(superType, value);
+		    if (enableTypeMappings){
+		//       	 conf.addLabelDecl(type,type,type.substring(0, 2),type.substring(0, 1));
+		   	    conf.addLabelDecl(value,value);
+		    }else {
+		     conf.addLabelDecl(anno.getType(), aFS.getType().getShortName(), aFS.getType()
+		            .getShortName().substring(0, 1));
+		    }
+		    if (!conf.hasDrawingDecl(anno.getType())) {
+		        conf.addDrawingDecl(new BratTextAnnotationDrawingDecl(anno.getType(), "black",
+		                palette[nextPaletteIndex % palette.length]));
+		        nextPaletteIndex++;
+		    }
+		    
+		    aDoc.addAnnotation(anno);
+
+	        // writeAttributes(anno, aFS);
```

Comment:
Why did you disable the ability to write attributes?

Reply:
They are not displayed, and I could not see any advantage  of having them